fuzzypp/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(fuzzyppcli VERSION 0.0.1 LANGUAGES CXX)
+
+# Set compile options. 
+add_compile_options(-O3)
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    add_compile_options(-Wall
+                        -Wextra
+                        -Werror
+                        -Wwrite-strings
+                        -Winit-self
+                        -Wcast-align
+                        -Wcast-qual
+                        -Wpointer-arith
+                        -Wstrict-aliasing
+                        -Wshadow
+                        -Wformat=2
+                        -Wmissing-declarations
+                        -Wmissing-include-dirs
+                        -Wno-unused-parameter
+                        -Wuninitialized
+                        -ansi
+                        -pedantic)
+endif()
+
+# Building with Clang is not currently supported.
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    add_compile_options(-Weverything
```

Comment:
I think this is a bit too much to handle. IIRC some warnings there even contradict the others.

Reply:
I need to play around with the Clang support. I don't think it compiles at the moment (not even due to having `-Weverything` turned on!)